packages/CopyInput/src/CopyInput.js

```diff
@@ -46,12 +48,22 @@ function CopyInput(props) {
   }
 
   return (
-    <sc.CopyInput data-pka-anchor="copy-input" {...moreProps}>
-      <Input ref={inputRef} defaultValue={value} isReadOnly={isReadOnly} {...extendedInputProps} />
+    <sc.CopyInput
+      data-pka-anchor="copy-input"
+      hasDefaultButtonBorder={!(hasValueShown && hasInputContainer)}
```

Comment:
I think `!hasValueShown && !hasInputContainer` would be easier to read, but I think if you take my suggestion below, this will be simpler anyway.